### PR TITLE
feat(gorup.search): Add RegEx Escaping

### LINKED ISSUE #27: Search group bug
When searching '[]', there is an error because we are entering the regex directly.

### PR DESCRIPTION
**Pull request by PCOffline**

### CODE REVIEW

**Comment by PCOffline** [comment]
Should I upload the updated docker-compose too?

File: src/group/endpoints/group.search.ts [hunk 1]
```diff
@@ -24,6 +24,12 @@ export default class SearchGroup extends Endpoint {
     });
   }
 
+  private static escapeRegex(str: string): string {
+    if (typeof str !== 'string') return str; // TODO: Decide whether this throws an error or not
```
Comment (carmeltsadok) [inline src/group/endpoints/group.search.ts:28]: The `partial` must be a string. It can also be a string that contains only numbers.

**Review by carmeltsadok — CHANGES_REQUESTED** [review]

**Comment by carmeltsadok** [comment]
I don't think you should update the `docker-compose` file because each one runs his file the way he wants to.
Also, I would call the branch `fix/regex-escape` instead of `feat/regex-escape` @PCOffline

**Comment by PCOffline** [comment]
> The `partial` must be a string. It can also be a string that contains only numbers. 

So that check is redundant?

**Pull request closed without merging**